.packit.yml

```
@@ -0,0 +1,18 @@
+---
+specfile_path: nmstate.spec
+upstream_package_name: nmstate
+upstream_project_url: http://nmstate.io
+create_tarball_command: ["python3", "setup.py", "sdist", "--dist-dir", "."]
+current_version_command: ["python3", "setup.py", "--version"]
+create_pr: true
+actions:
+  post-upstream-clone: "bash -c './packaging/make_spec.sh > nmstate.spec'"
+jobs:
+  - job: copr_build
+    trigger: pull_request
+    metadata:
+      targets:
+        - centos-stream-x86_64
+        - epel-8-x86_64
+        - fedora-32-x86_64
+        - fedora-rawhide-x86_64
```

You can also add the `tests` job (see [the relevant part of the docs](https://packit.dev/testing-farm/) for more info.

- If you only add the job to the config, it will try to install the package from the created copr repo. You can also set up real tests using tmt/fmf.
- This should be enough (the targets will be copied from the build):

```suggestion
        - fedora-rawhide-x86_64
- job: tests
  trigger: pull_request
```

---

To install it on CentOS, it currently needs extra Copr repos for NetworkManager and maybe also openvswitch. Is it possible to specify them?
However, currently the possibility to get Copr repos for pull requests to simplify testing of PRs is the main reason we added Packit, so this is not a a priority.

---

> To install it on CentOS, it currently needs extra Copr repos for NetworkManager and maybe also openvswitch. Is it possible to specify them?

- It's currently not possible, but a good suggestion. I'll create an issue for that.
- We can build in the custom copr project, but that's probably not what you want.

---

https://github.com/packit-service/packit-service/issues/437

---

Please see the issue, I believe that could work for this case .... prepare happens before artifact installation ... we now support simple shell steps ....